### DISCUSSION TITLE
probabilisticsamplerprocessor: should sampling.priority drop Spans

### DISCUSSION THREAD

**Discussion by shawnsong** [open-telemetry/opentelemetry-collector-contrib#10438] [Q&A] [answered]
### Context

When adding `probabilisticsamplerprocessor` to our collector, I realized that all Spans with `sampling.priority` set to `0` are dropped by the processor. 

We use a parent-based sampler in our application code and we add `sampling.priority=0` to all Spans by default to prevent Spans from starting from everywhere (for example we don't want any traces for health checks and some scheduled jobs). 

Is it possible to have this processor respect the decision of a Span's parent if it exists?

**Comment by jpkrohling** [comment, marked as answer]
Sorry, I'm not sure I understand it. Are you saying that this processor isn't dropping spans with sampling.priority=0? Can you share your config and a couple of spans, with your expectation on what should happen to them? Or better yet, can you create a unit test demonstrating the problem? The suite does seem to have tests covering this case, but I'm really interested in this if you can provide a reproducer.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/3ba302b4fd7b16055116217d34cc19e2fb38f176/processor/probabilisticsamplerprocessor/probabilisticsampler_test.go#L270-L298
